README.md

```diff
@@ -275,22 +434,70 @@ The `Config` object describes which components Puck should render, how they shou
+    - **resolveProps** (`async (props: object) => object` [optional]): Function to dynamically change props before rendering the component.
+      - Args
+        - **props** (`object`): the current props for your component stored in the Puck data
+      - Response
+        - **props** (`object`): the resolved props for your component. Will not be stored in the Puck data
+        - **readOnly** (`object`): an object describing which fields on the component are currently read-only
+          - **[prop]** (`boolean`): boolean describing whether or not the prop field is read-only
 
 ### `Field`
 
 A `Field` represents a user input field shown in the Puck interface.
 
-- **type** (`text` | `textarea` | `number` | `select` | `radio` | `external` | `array` | `custom`): The input type to render
+### All Fields
+
 - **label** (`text` [optional]): A label for the input. Will use the key if not provided.
-- **arrayFields** (`object`): Object describing sub-fields for items in an `array` input
-  - **[fieldName]** (`Field`): The Field objects describing the input data for each item
-- **getItemSummary** (`(object, number) => string` [optional]): Function to get the name of each item when using the `array` or `external` field types
-- **defaultItemProps** (`object` [optional]): Default props to pass to each new item added, when using a `array` field type
-- **options** (`object[]`): array of items to render for select or radio inputs
+
+### Text Fields
+
+- **type** (`"text"`)
+
+### Textarea Fields
+
+- **type** (`"textarea"`)
+
+### Number Fields
+
+- **type** (`"number"`)
+
+### Select Fields
+
+- **type** (`"select"`)
+- **options** (`object[]`): array of items to render
+  - **label** (`string`)
+  - **value** (`string` | `number` | `boolean`)
+
+### Radio Fields
+
+- **type** (`"radio"`)
+- **options** (`object[]`): array of items to render
   - **label** (`string`)
   - **value** (`string` | `number` | `boolean`)
-- **adaptor** (`Adaptor`): Content adaptor if using the `external` input type
+
+### Array Fields
+
+- **type** (`"array"`)
+- **arrayFields** (`object`): Object describing sub-fields for each item
+  - **[fieldName]** (`Field`): The Field objects describing the input data for each item
+  - **getItemSummary** (`(object, number) => string` [optional]): Function to get the label of each item
+- **defaultItemProps** (`object` [optional]): Default props to pass to each new item added, when using a `array` field type
+
+### External Fields
+
+External fields can be used to load content from an external content repository, like Strapi.js, using an `Adaptor`.
+
+- **type** (`"external"`)
+- **adaptor** (`Adaptor`): Content adaptor responsible for fetching data to show in the table
+  - **name** (`string`): The human-readable name of the adaptor
+  - **fetchList** (`(adaptorParams: object) => object`): Fetch a list of content and return an array
```

Comment:
Could Adaptors fetch an object instead of a list?

Reply:
They could fetch an object but this method needs to return an array to populate the table

Reply:
Wonder if "a list of content" is correct in that case? Maybe `Fetch ~a list of~ content and return an array`? I could be missing the point here 😊

Reply:
I've updated to `Fetch content from a third-party API and return an array` for clarity